signing/signing.go

```diff
@@ -0,0 +1,106 @@
+package signing
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/ossf/scorecard-action/entrypoint"
+	"github.com/ossf/scorecard-action/options"
+	sigOpts "github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/sign"
+)
+
+func SignScorecardResult(scorecardResultsFile string) error {
+	if err := os.Setenv("COSIGN_EXPERIMENTAL", "true"); err != nil {
+		return fmt.Errorf("error setting COSIGN_EXPERIMENTAL env var: %w", err)
+	}
+
+	// Prepare settings for SignBlobCmd.
+	rootOpts := &sigOpts.RootOptions{Timeout: sigOpts.DefaultTimeout} // Just the timeout.
+
+	keyOpts := sign.KeyOpts{
+		FulcioURL:    sigOpts.DefaultFulcioURL,     // Signing certificate provider.
+		RekorURL:     sigOpts.DefaultRekorURL,      // Transparency log.
+		OIDCIssuer:   sigOpts.DefaultOIDCIssuerURL, // OIDC provider to get ID token to auth for Fulcio.
+		OIDCClientID: "sigstore",
+	}
+	regOpts := sigOpts.RegistryOptions{} // Not necessary so we leave blank.
+
+	// This command will use the provided OIDCIssuer to authenticate into Fulcio, which will generate the
+	// signing certificate on the scorecard result. This attestation is then uploaded to the Rekor transparency log.
+	// The output bytes (signature) and certificate are discarded since verification can be done with just the payload.
+	if _, err := sign.SignBlobCmd(rootOpts, keyOpts, regOpts, scorecardResultsFile, true, "", ""); err != nil {
+		return fmt.Errorf("error signing payload: %w", err)
+	}
+
+	return nil
+}
+
+// Changes output settings to json and runs scorecard again.
+// TODO: run scorecard only once and generate multiple formats together.
+func GetJsonScorecardResults() ([]byte, error) {
+	defer os.Setenv(options.EnvInputResultsFile, os.Getenv(options.EnvInputResultsFile))
+	defer os.Setenv(options.EnvInputResultsFormat, os.Getenv(options.EnvInputResultsFormat))
+	os.Setenv(options.EnvInputResultsFile, "results.json")
+	os.Setenv(options.EnvInputResultsFormat, "json")
+
+	actionJson, err := entrypoint.New()
+
+	if err != nil {
+		return nil, fmt.Errorf("creating scorecard entrypoint: %v", err)
+	}
+	if err := actionJson.Execute(); err != nil {
+		return nil, fmt.Errorf("error during command execution: %v", err)
+	}
+
+	// Get json output data from file.
+	jsonPayload, err := ioutil.ReadFile(os.Getenv(options.EnvInputResultsFile))
+	if err != nil {
+		return nil, fmt.Errorf("reading scorecard json results from file: %v", err)
+	}
+
+	return jsonPayload, nil
+}
+
+// Calls scorecard-api to process & upload signed scorecard results.
+func ProcessSignature(sarifPayload []byte, jsonPayload []byte, repoName string, repoRef string) error {
```

Comment:
small suggestion! with go you can combine type params if the subsequent ones are the same.. e.g.:
```
ProcessSignature(sarifPayload, jsonPayload []byte, repoName, repoRef string)
```